### DISCUSSION TITLE
Robot Inventor Hub - firmware updated

### DISCUSSION THREAD

**Discussion by alpha71** [pybricks/support#331] [Q&A]
Hello,
i'd like to understand 
i installed the firmware on the Inventor hub thru the website beta.pybricks.com , then i tried to connect to the code page (code.pybricks.com) via BT but the hub cannot be found! (i used Chrome, actually i already succesfully connected the city hub and technics hub!)
Now i'm not clear if the opration of the firmware is completed or still miss some steps???
Let me know the exaclty steps please, thank you.
Alessandro

**Comment by dlech** [comment]
The firmware is dual-boot, so it can either boot the official LEGO firmware (light is white) or if you hold the power button longer, it will boot the Pybricks firmware (light is blue).

**Comment by alpha71** [comment]
Perfect! It works fine. Another question: when download a programs into the hub it can be saved inside the Eprom (like for the official firmware which can store up to 10 programs!) or the program just start by the Pybrick code?
Thank you again

**Reply by dlech** [reply]
We don't have the external flash memory working yet. I don't recall if any of our tools support this with the dual-boot firmware at this time, but it is theoretically possible to include one program when you flash the firmware.

**Reply by alpha71** [reply]
Hello David, if I understood well, this means that the hub cannot work independently from PC like EV3, or better the program has to be lunched however by PC then will constantly make communication via BT, instead all tasks will be operated by the hub itself even if the PC is disconnected (power off or whatever!), is it correct? If like this compared with the official firmware brings some limitation isn't?

**Reply by laurensvalk** [reply]
Yes, that is true, *for now* 😉

We will make it possible to save files later. We are a small team, so it takes a while to implement all features.

**Reply by alpha71** [reply]
Yeah, I can understand it, I think all of you already done a great job. At the moment I'm focus more in the EV3 with Micro-python which actually I'm comparing with NXC on the NTX (I found Micro-python is really easier and powerful). The target is to have a unique programming language which can offer the best compromise with simplicity and efficiency, and I think Python is in the right direction.

**Reply by alpha71** [reply]
Hello guys, i actually in Code.Pybricks i tried to flash the firmware to have atleast one program saved (as David wrote already!) but seems cannot find the BT connection (always searching!), instead i've no pbm to load the program. So there is a provedure to do so? I also read in some issues where it was mentioned a procedure to flash the firmware and the program but i'm lost on that!
I know the Inventor and Prime Hub are still in Beta version but i'd like to understand if i'm doing something wrong or just it's so for the moment!
Another point regarding the way to write programs, there is a chance to use VisualCode? I know Python version in Prime is 3.0 instead the 2.0 of EV3, but i'd like to know if can be implemented or not? Or probably it's meanless!

**Reply by laurensvalk** [reply]
The Inventor Hub / SPIKE Prime Hub do not support a permanent program on the hub yet. Its file storage works a bit differently.
